proposals/fate-operator-proposal.md

```diff
@@ -0,0 +1,220 @@
+4. [Utilization of FATE in Anti Money Laundering Through Multiple Banks](https://www.fedai.org/cases/utilization-of-fate-in-anti-money-laundering-through-multiple-banks/)
+
+Other Federated Machine Learning cases:
+1. [Federated Learning for Mobile Keyboard Prediction](https://research.google/pubs/pub47586/)
+2. [Federated Learning powered by NVIDIA Clara](https://devblogs.nvidia.com/federated-learning-clara/): hospitals and medical institutions collaboratively share and combine their local knowledge
+3. [Owkin Launches the Collaborative COVID-19 Open AI Consortium (COAI)](https://www.unite.ai/covid-19-open-ai-consortium/)
+
+## Motivation
+Kubeflow provides a toolset for end-to-end machine learning workflow on Kubernetes. Introducing the capability of federated learning to Kubeflow helps FML users and researchers leverage existing Kubeflow toolkits in their workflows and help them more efficiently build FML solutions. 
+
+A FATE-Operator is a start of supporting FML in Kubeflow. This proposal is aimed to defining what FATE operator should look like, and how to apply to Kubeflow.
+
+## Goals
+A Kubeflow user should be able to run training using FATE as easily as they can using PyTorch, Tensorflow. This proposal is centered around a Kubernetes Operator for FATE. With the FATE-Operator, a user can:
+1.	Provision and manage a FATE cluster;
+2.	Submit an FML job to FATE.
+
+This proposal defines the following:
+1.	A FATE operator with three CRDs: 
+   * FateJob: create an FML job;
+   * FateCluster: create a FATE cluster to serve FML jobs;
+   * Kubefate: the resource management component of FATE cluster.
+2.	Example of full lifecycle to create KubeFATE component, deploy FATE cluster and submit an FML job to created FATE and get the result. Note that, KubeFATE and FATE cluster only needs to be deployed once, and can handle multiple jobs.
+
+## Non-Goals
+For the scope of this proposal, we won’t be addressing the method of serving the model.
+
+## Design
+
+### Container images and deploying FATE cluster on Kubernetes
+We have built a set of Docker images for FATE cluster, and put into: https://hub.docker.com/orgs/federatedai/repositories . All the images have been already used and verified by community users. 
+
+There is a provisioning and management component of FATE cluster, called [KubeFATE](https://github.com/FederatedAI/KubeFATE/tree/master/k8s-deploy). KubeFATE manages FATE clusters of one party in a federation.
+
+All images work well and are proven in users’ environments.
+
+### Custom Resource Definition
+#### Kubefate
+```
+apiVersion: app.kubefate.net/v1beta1
+kind: Kubefate
+metadata:
+  name: kubefate-sample
+  namespace: kube-fate
+spec:
+  # kubefate image tag
+  imageVersion: v1.0.2
+  # ingress host
+  ingressDomain: kubefate.net
+  # kubefate config
+  config:
+    FATECLOUD_MONGO_USERNAME: "root"
```

Comment:
Are alternative databases to MONGO possible? I think MONGO changed their license and is no longer a widely accepted OSS license like Apache2.  This might limit how closely we can integrate the two. For example, it might prevent us from including Fate in our standard configurations or hosting those images in our repos.

Also is it possible to use secrets to store the password and not put it in the config. 

/cc @animeshsingh

Reply:
Yes. It has been changed to secrets. I will update the CRD sample in proposal right now.

Reply:
@LaynePeng Apache CouchDB can be a more elegant choice, given its already in Apache foundation
https://couchdb.apache.org/